frontend/lib/resolution/InitResolver.cpp

```diff
@@ -341,15 +342,13 @@ const Type* InitResolver::computeReceiverTypeConsideringState(void) {
   }
 
   if (subs.size() == 0) {
-    ctx_->error(ctInitial->id(), "unable to instantiate generic type from initializer");
     return currentRecvType_;
+  } else {
+    const Type* ret = ctFromSubs(ctx_, initialRecvType_, ctInitial, subs);
+    CHPL_ASSERT(ret);
+    return ret;
```

Comment:
Personally, I prefer the form in which this is not an `else` clause (since the `if` already returns). But if you feel strongly enough to keep it, that's fine with me too.

Reply:
I could see either way, but would rather leave this version. My logic is I prefer an explicit else branch over an if with an early exit, unless that if is an unusual/error condition (subjectively). I changed it since I don't see the if as an error condition after the change.